src/vcModals.cpp

```diff
@@ -656,10 +656,89 @@ void vcModals_DrawProjectInfo(vcState *pProgramState)
     if (ImGui::BeginChild("##infoModal", ImVec2(-1, -30), true))
     {
       if (udProjectNode_GetMetadataString(pProgramState->activeProject.pRoot, "information", &pInfo, "") == udE_Success)
-        ImGui::TextWrapped("%s", pInfo);
+      {
+        if (udStrcmp(pProgramState->projectInfoTextures.pLastInfoText, pInfo) != 0)
+        {
+          pProgramState->projectInfoTextures.pLastInfoText = udStrdup(pInfo);
+          for (vcTexture *pTexture : pProgramState->projectInfoTextures.textures)
+            vcTexture_Destroy(&pTexture);
+          for (const char *pStr : pProgramState->projectInfoTextures.infoStrings)
+            udFree(pStr);
+          pProgramState->projectInfoTextures.infoStrings.Clear();
+          pProgramState->projectInfoTextures.textures.Clear();
+          pProgramState->projectInfoTextures.textureSizes.Clear();
+          
+          // Discover Images
+          int firstIndex = 0;
+          int startStrIndex = 0;
+          while (true)
+          {
+            size_t startImagePos = 0;
+            size_t openBracketPos = 0;
+            size_t closeBracketPos = 0;
+            size_t openParanthesisPos = 0;
+            size_t closeParanthesisPos = 0;
+
+            udStrchr(pInfo + firstIndex, "!", &startImagePos);
+            startImagePos += firstIndex;
+            udStrchr(pInfo + startImagePos, "[", &openBracketPos);
+            openBracketPos += startImagePos;
+            udStrchr(pInfo + openBracketPos, "]", &closeBracketPos);
+            closeBracketPos += openBracketPos;
+            udStrchr(pInfo + closeBracketPos, "(", &openParanthesisPos);
+            openParanthesisPos += closeBracketPos;
+            udStrchr(pInfo + openParanthesisPos, ")", &closeParanthesisPos);
+            closeParanthesisPos += openParanthesisPos;
+            
+            const int64_t invalidLen = udStrlen(pInfo);
```

Comment:
This operation isn't free, you execute this N+1 times, where N is the number of valid `![alt](url)` patterns. However, it won't change. Additionally, `udStrchr` returns `nullptr` when the character isn't found which would allow you to exit early more often.

Reply:
moved outside while loop